modules/secure-logging/slog.c

```diff
@@ -420,15 +424,34 @@ gchar *convertToBase64(unsigned char *input, gsize len)
  */
 void cmac(unsigned char *key, const void *input, gsize length, unsigned char *out, gsize *outlen)
 {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+  EVP_MAC *mac = EVP_MAC_fetch(NULL, "CMAC", NULL);
+  OSSL_PARAM params[] =
+  {
+    OSSL_PARAM_utf8_string("cipher", "aes-256-cbc", 0),
+    OSSL_PARAM_END,
+  };
+
+  EVP_MAC_CTX *ctx = EVP_MAC_CTX_new(mac);
+
+  EVP_MAC_init(ctx, key, KEY_LENGTH, params);
+  EVP_MAC_update(ctx, input, length);
+  size_t out_len;
+  EVP_MAC_final(ctx, out, &out_len, 99999);
+
+  EVP_MAC_CTX_free(ctx);
+  EVP_MAC_free(mac);
+#else
   CMAC_CTX *ctx = CMAC_CTX_new();
 
   CMAC_Init(ctx, key, KEY_LENGTH, EVP_aes_256_cbc(), NULL);
   CMAC_Update(ctx, input, length);
 
-  size_t outsize;
-  CMAC_Final(ctx, out, &outsize);
-  *outlen = outsize;
+  size_t out_len;
+  CMAC_Final(ctx, out, &out_len);
+  *outlen = out_len;
```

Comment:
[nitpicking, no need to change anything]

Reformatting lines of code, which we did not modify could go to separate commits. Same applies for removing dead code (`sLogDecrypt()` in the header file).

Reply:
Because crypto-related code is modified here, and we almost missed a bug in the few lines above, I think it would be best to split this commit to make it as atomic and clean as possible.